Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=7460db567bbca76bf087d1694d792a1a96bdaa26
Commit:     7460db567bbca76bf087d1694d792a1a96bdaa26
Parent:     7bcc1bb1232de6efc0b85e0c7fe38e90b2436318
Author:     Magnus Damm <damm@igel.co.jp>
AuthorDate: Thu Jan 29 14:25:12 2009 -0800
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Thu Jan 29 18:04:43 2009 -0800

    gpiolib: fix request related issue
    
    Fix request-already-requested handling in gpio_request().
    
    Signed-off-by: Magnus Damm <damm@igel.co.jp>
    Acked-by: David Brownell <dbrownell@users.sourceforge.net>
    Cc: <stable@kernel.org>		[2.6.28.x]
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

 drivers/gpio/gpiolib.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/gpio/gpiolib.c b/drivers/gpio/gpiolib.c
index 35e7aea..42fb2fd 100644
--- a/drivers/gpio/gpiolib.c
+++ b/drivers/gpio/gpiolib.c
@@ -789,6 +789,7 @@ int gpio_request(unsigned gpio, const char *label)
 	} else {
 		status = -EBUSY;
 		module_put(chip->owner);
+		goto done;
 	}
 
 	if (chip->request) {
